### PR TITLE
Update Checkstyle to 8.1

### PR DESCRIPTION
**Pull request by AustinShalit**
Also added some new checks.

### CODE REVIEW

File: checkstyle.xml [hunk 1]
```diff
@@ -82,6 +82,11 @@
         <module name="ArrayTypeStyle"/>
         <module name="MissingSwitchDefault"/>
         <module name="FallThrough"/>
+        <module name="SimplifyBooleanExpression"/>
+        <module name="SimplifyBooleanReturn"/>
+        <module name="StringLiteralEquality"/>
+        <module name="UnnecessaryParentheses"/>
+        <module name="ExplicitInitializationCheck"/>
```
Comment (SamCarlberg) [inline checkstyle.xml]: This was deliberately left out so we can be explicit with initializations. It's easy to forget that booleans default to `false`, for example.

Comment (AustinShalit) [inline checkstyle.xml]: From: http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/checks/coding/ExplicitInitializationCheck.html

>Rationale: each instance variable gets initialized twice, to the same value. Java initializes each instance variable to its default value (0 or null) before performing any initialization specified in the code. So in this case, x gets initialized to 0 twice, and bar gets initialized to null twice. So there is a minor inefficiency. This style of coding is a hold-over from C/C++ style coding, and it shows that the developer isn't really confident that Java really initializes instance variables to default values.

Comment (SamCarlberg) [inline checkstyle.xml]: >  So there is a minor inefficiency

Which is completely irrelevant. The code is more readable if you can go to the declaration and see what the initial value is *immediately* without having to remember or google what the default value is

File: src/main/java/edu/wpi/first/shuffleboard/sources/NetworkTableSource.java [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
   protected final String fullTableKey;
   private int listenerUid = -1;
-  private volatile boolean ntUpdate = false;
+  private volatile boolean ntUpdate;
```
Comment (nightpool) [inline src/main/java/edu/wpi/first/shuffleboard/sources/NetworkTableSource.java]: should be undone

File: src/main/java/edu/wpi/first/shuffleboard/sources/recording/Playback.java [hunk 3]
```diff
@@ -41,7 +41,7 @@
   private final int numFrames;
   private final int maxFrameNum;
   private Thread autoRunner;
-  private volatile boolean started = false;
+  private volatile boolean started;
```
Comment (nightpool) [inline src/main/java/edu/wpi/first/shuffleboard/sources/recording/Playback.java]: ibid

File: src/main/java/edu/wpi/first/shuffleboard/widget/BooleanBox.java [hunk 4]
```diff
@@ -43,7 +43,7 @@ private Color getColor() {
     final Boolean data = getData();
     if (data == null) {
       return Color.BLACK;
-    } else if (data == true) { //NOPMD
+    } else if (data) {
```
Comment (nightpool) [inline src/main/java/edu/wpi/first/shuffleboard/widget/BooleanBox.java]: was NOPMD for a reason

File: src/main/java/edu/wpi/first/shuffleboard/util/Maps.java [hunk 5]
```diff
@@ -32,7 +32,7 @@ private Maps() {
    */
   public static final class MapBuilder<K, V> {
 
-    private boolean completed = false;
+    private boolean completed;
```
Comment (nightpool) [inline src/main/java/edu/wpi/first/shuffleboard/util/Maps.java]: same as above

File: src/main/java/edu/wpi/first/shuffleboard/widget/WidgetType.java [hunk 6]
```diff
@@ -7,11 +7,15 @@
 
 public interface WidgetType extends Supplier<Widget> {
   /**
+   * Get the name of the widget.
+   *
```
Comment (nightpool) [inline src/main/java/edu/wpi/first/shuffleboard/widget/WidgetType.java]: this is way too verbose, and just clutters up the code for no reason. Either remove the `@returns` statement or leave it as the only doc.

File: src/main/java/edu/wpi/first/shuffleboard/widget/WidgetType.java [hunk 7]
```diff
@@ -7,11 +7,15 @@
...
    * @return The name of the widget.
    */
   String getName();
 
   /**
+   * Get the data types the widget should be suggested for.
+   *
```
Comment (nightpool) [inline src/main/java/edu/wpi/first/shuffleboard/widget/WidgetType.java]: ibid

File: src/main/java/edu/wpi/first/shuffleboard/widget/Widgets.java [hunk 8]
```diff
@@ -22,7 +22,7 @@
 public final class Widgets {
 
   private static Map<String, WidgetType> widgets = new TreeMap<>();
-  private static boolean loadedStockWidgets = false;
+  private static boolean loadedStockWidgets;
```
Comment (nightpool) [inline src/main/java/edu/wpi/first/shuffleboard/widget/Widgets.java]: booleans should be explicitly initialized

File: src/main/java/edu/wpi/first/shuffleboard/components/WidgetPane.java [hunk 9]
```diff
@@ -151,7 +151,9 @@ public WidgetTile addWidget(Widget widget, GridPoint location, TileSize size) {
   }
 
   /**
-   * @return Returns the expected size of the widget, in tiles.
+   * Gets the expected size of the widget.
+   *
+   * @return the expected size of the widget, in tiles.
```
Comment (nightpool) [inline src/main/java/edu/wpi/first/shuffleboard/components/WidgetPane.java]: redundant.

File: checkstyle.xml (hunk 1 above)
Comment (nightpool) [inline checkstyle.xml]: If we're not initializing an object value in the constructor, then that deserves to be called out and made explicit in the code using ` = null`. 

Booleans should *always* be explicitly initialized—the idea that one boolean type is "default" is an anachronistic holdover from C and not at all obvious to most people.

**Comment by AustinShalit** [comment]
@nightpool Please review.

**Pull request merged**